Lib/email/message.py

```diff
@@ -290,7 +290,7 @@ def get_payload(self, i=None, decode=False):
         cte = str(self.get('content-transfer-encoding', '')).lower()
         # payload may be bytes here.
         if isinstance(payload, str):
-            if utils._has_surrogates(payload):
+            if utils._has_decoded_with_surrogateescape(payload):
                 bpayload = payload.encode('ascii', 'surrogateescape')
```

Comment:
`payload.encode('ascii', 'surrogateescape')` is calculated twice. First in `_has_decoded_with_surrogateescape()`, the, if it successes, the result is dropped and it is encoded again here. It is possible to optimize this, but it requires larger change.